pentagon/component/gcp/__init__.py

```diff
@@ -0,0 +1,132 @@
+
+    def get(self, destination):
+        service = build('container', 'v1')
+        cl = service.projects().zones().clusters()
+        self._zone = self._data.get('zone')
+        self._project = self._data.get('project')
+        self._destination = destination
+        cluster_name = self._data.get('name')
+        if cluster_name:
+            cluster = cl.get(projectId=self._project, zone=self._zone, clusterId=cluster_name).execute()
+            Cluster(self.parse_cluster(cluster)).add(cluster_name)
+            self.add_cluster_nodepools(cluster_name, cluster.get('nodePools'))
+        else:
+            clusters = cl.list(projectId=self._project, zone=self._zone).execute()
+
+            for cluster in clusters['clusters']:
+                Cluster(self.parse_cluster(cluster)).add(cluster['name'])
+                self.add_cluster_nodepools(cluster.get('name'), cluster.get('nodePools'), self._data.get('get_default_nodepools'))
+
+    def parse_cluster(self, cluster):
+        """ parses cluster object returned from api call and converts it to a data strucutur for the .add() method """        
+
+        # remove master zone from additional-zones
+        locations = cluster.get('locations')
+        locations.remove(self._zone)
+
+        for np in cluster.get('nodePools'):
+            if np.get('name') == 'default-pool':
+                default_nodepool = np
+
+        return {
+                    'project': self._project,
+                    'zone': self._zone,
+                    'cluster_name': cluster['name'],
+                    'cluster_cidr': cluster['clusterIpv4Cidr'],
+                    'cluster_version': cluster['currentMasterVersion'],
+                    'additional_zones': locations,
+                    'disk_size': cluster['nodeConfig']['diskSizeGb'],
+                    'max_nodes_per_pool': cluster.get('maxNodesPerPool', None),
+                    'image_type': cluster['nodeConfig']['imageType'],
+                    'node_labels': cluster.get('nodeLabels', []),
+                    'num_nodes': cluster['currentNodeCount'],
+                    'image_type': cluster['nodeConfig']['imageType'],
+                    'username': cluster['masterAuth']['username'],
+                    'network': cluster['network'],
+                    'subnetwork': cluster.get('subnetwork', None),
+                    'scopes': cluster['nodeConfig']['oauthScopes'],
+                    'enable_autoscaling': np.get('autoscaling'),
+                    'min_nodes': np.get('autoscaling', {}).get('minNodeCount'),
+                    'max_nodes': np.get('autoscaling', {}).get('maxNodeCount'),
+                }
+
+    def add_cluster_nodepools(self, cluster_name, nodepools, get_default_nodepools=False):
+        """ parses cluster node pools and adds nodepool objects """
+        for nodepool in nodepools:
+            if nodepool['name'] != 'default-poo_namel' or get_default_nodepools:
```

Comment:
Should this be `default-pool` or `default-pool_name` maybe? Checking a cluster in GKE it appears to have the name `default-pool`.